include/RAJA/util/AllocatorUmpire.hpp

```diff
@@ -0,0 +1,113 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   RAJA header file containing an implementation of a memory pool.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-21, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_UTIL_ALLOCATORUMPIRE_HPP
+#define RAJA_UTIL_ALLOCATORUMPIRE_HPP
+
+#include <string>
+
+#include "RAJA/util/Allocator.hpp"
+
+namespace RAJA
+{
+
+/*! \class AllocatorUmpire
+ ******************************************************************************
+ *
+ * \brief  AllocatorUmpire wraps an umpire allocator so it can be used in RAJA.
+ *         This is intended to be used with umpire memory pools.
+ *
+ * Example:
+ *
+ * umpire::Allocator umpire_allocator;
+ *
+ * RAJA::AllocatorUmpire<umpire::Allocator> raja_alocator(umpire_allocator);
+ *
+ ******************************************************************************
+ */
+template <typename allocator_t>
+struct AllocatorUmpire : Allocator
```

Comment:
This class is designed to wrap an umpire allocator to make it easy for codes to replace RAJA's allocators with umpire pools.